components/properties/src/script.rs

```diff
@@ -133,13 +165,84 @@ impl<'data> ScriptExtensions<'data> {
             let scx_val = self.extensions.get(ext_idx as usize);
             scx_val.unwrap_or_default()
         } else {
-            let script_with_ext_ule = self.trie.get_ule(code_point);
-            let script_ule_slice = script_with_ext_ule
-                .map(core::slice::from_ref)
-                .unwrap_or_default();
+            // Note: `Script` and `ScriptWithExt` are both represented as the same
+            // u16 value when the `ScriptWithExt` has no higher-order bits set.
+            let script_ule_slice = core::slice::from_ref(sc_with_ext_ule);
             ZeroSlice::from_ule_slice(script_ule_slice)
         }
     }
+
+    /// Return the `Script_Extensions` property value for this code point.
```

Comment:
Nit: Please add an example to the public methods on ScriptExtensions:

- get_script_val
- get_script_extensions_val
- has_script
- get_script_extensions_ranges
- get_script_extensions_set

You can do this in a follow-up if you prefer.